client/src/components/Form/Elements/FormSelect.vue

```diff
@@ -0,0 +1,182 @@
+}
+
+const props = withDefaults(
+    defineProps<{
+        multiple?: boolean;
+        optional?: boolean;
+        options: Array<[string, string]>;
+        value?: Array<string> | string;
+    }>(),
+    {
+        multiple: false,
+        optional: false,
+        value: null,
+    }
+);
+
+const emit = defineEmits<{
+    (e: "input", value: SelectValue | Array<SelectValue>): void;
+}>();
+
+/**
+ * Determine dom wrapper class
+ */
+const cls: ComputedRef<string> = computed(() => {
+    return props.multiple ? "form-select-multiple" : "form-select-single";
+});
+
+/**
+ * Configure deselect label
+ */
+const deselectLabel: ComputedRef<string> = computed(() => {
+    return props.multiple ? "Click to remove" : "";
+});
+
+/**
+ * Translates input options for consumption by the
+ * select component into an array of objects
+ */
+const formattedOptions: ComputedRef<Array<SelectOption>> = computed(() => {
+    const result: Array<SelectOption> = props.options.map((option: [string, string]) => ({
+        label: option[0],
+        value: option[1],
+    }));
+    if (props.optional && !props.multiple) {
+        result.unshift({
+            label: "Nothing selected",
+            value: null,
+        });
+    }
+    return result;
+});
+
+/**
+ * Tracks if the select field has options
+ */
+const hasOptions: ComputedRef<Boolean> = computed(() => {
+    return formattedOptions.value.length > 0;
+});
+
+/**
+ * Configure selected label
+ */
+const selectedLabel: ComputedRef<string> = computed(() => {
+    return props.multiple ? "Selected" : "";
+});
+
+/**
+ * Tracks selected values
+ */
+const selectedValues: ComputedRef<Array<SelectValue>> = computed(() => {
+    return Array.isArray(props.value) ? props.value : [props.value];
+});
+
+/**
+ * Tracks current value and emits changes
+ */
+const currentValue = computed({
+    get: () => formattedOptions.value.filter((option: SelectOption) => selectedValues.value.includes(option.value)),
+    set: (val: Array<SelectOption> | SelectOption): void => {
+        if (Array.isArray(val)) {
+            if (val.length > 0) {
+                const values: Array<SelectValue> = val.map((v: SelectOption) => v.value);
+                emit("input", values);
+            } else {
+                emit("input", null);
+            }
+        } else {
+            emit("input", val.value);
+        }
+    },
+});
+
+/**
+ * Ensures that an initial value is selected for non-optional inputs
+ */
+function setInitialValue(): void {
+    if (props.value === null && !props.optional && hasOptions.value) {
+        const initialValue = formattedOptions.value[0];
+        if (initialValue) {
+            emit("input", initialValue.value);
+        }
+    }
+}
+
+/**
+ * Watches changes in select options and adjusts initial value if necessary
+ */
+watch(
+    () => props.options,
+    () => setInitialValue()
+);
+
+/**
+ * Sets initial value if necessary
+ */
+onMounted(() => {
+    setInitialValue();
```

Comment:
It doesn't look like we need DOM elements to set the initial value ... in fact couldn't initialValue be a normal computed ?

Reply:
I moved `initialValue` to a computed prop. However, we should keep the mounted and options watcher as triggers to set the initialValue instead of attempting to embed it into the currentValue computed prop. Checking and always forcing to reset the `initialValue` after initialization is imo confusing to the user and can lead to computed side effects. It is better for now to set the `initialValue` just once and then let the user interact with the interface. If the user unselects the `initialValue` a warning alert is highlighted as part of the regular form validation.